exec/cnex/lib/runtime_win32.c

```diff
@@ -0,0 +1,389 @@
+#include "runtime.h"
+
+#define COBJMACROS
+
+#include <iso646.h>
+#include <windows.h>
+
+#include "assert.h"
+#include "cell.h"
+#include "enums.h"
+#include "exec.h"
+#include "object.h"
+#include "stack.h"
+
+#ifdef _WIN32
+#include <io.h>
+#define strdup _strdup
+#else
+#include <unistd.h>
+#endif
+
+
+Object *com_createComObject(char *name, IDispatch *pdisp);
+
+typedef struct tagTComObject {
+    char *name;
+    IDispatch *obj;
+} ComObject;
+
+// ToDo: Implement UTF8 strings!
+BSTR bstr_from_utf8string(const char *s)
+{
+    int nchars = MultiByteToWideChar(CP_ACP, 0, s, -1, NULL, 0);
+    OLECHAR *wstr = malloc(sizeof(OLECHAR) * nchars);
+    MultiByteToWideChar(CP_ACP, 0, s, -1, wstr, nchars);
+    BSTR bstr = SysAllocString(wstr);
+    free(wstr);
+    return bstr;
+}
+
+// ToDo: Implement UTF8 strings!
+char *utf8string_from_bstr(BSTR bstr)
+{
+    if (bstr == NULL) {
+        return "null";
+    }
+    int nchars = WideCharToMultiByte(CP_ACP, 0, bstr, -1, NULL, 0, NULL, NULL);
+    char *buf = malloc(nchars);
```

Comment:
Memory leak here (not freed).

Reply:
It is free'd by the caller.  Fixed an instance where it wasn't free'ed.